source/transactions-convenient-api/transactions-convenient-api.md

```diff
@@ -0,0 +1,384 @@
+take advantage of this helper method.
+
+### Majority write concern is used when retrying commitTransaction
+
+Drivers should apply a majority write concern when retrying commitTransaction to guard against a transaction being
+applied twice. This requirement was originally enforced in the implementation of `withTransaction`, but will now be
+handled by the transaction spec itself in order to benefit applications irrespective of whether they use
+`withTransaction` (see the corresponding section in the
+[Transactions spec Design Rationale](../transactions/transactions.md#majority-write-concern-is-used-when-retrying-committransaction)).
+
+### The callback function has a flexible signature
+
+An original design considered requiring the callback to accept a ClientSession as its first parameter. That could be
+superfluous for languages where the callback might already have access to ClientSession through its lexical scope.
+Instead, the spec simply requires that drivers ensure the callback will be able to access the ClientSession.
+
+Similarly, the specification does not concern itself with the return type of the callback function. If drivers allow the
+callback to return a value, they may also choose to propagate that value as the return value of withTransaction.
+
+An earlier design also considered using the callback's return value to indicate whether control should break out of
+`withTransaction` (and its retry loop) and return to the application. The design allows this to be accomplished in one
+of two ways:
+
+- The callback aborts the transaction directly and returns to `withTransaction`, which will then return to its caller.
+- The callback raises an error without the "TransientTransactionError" label, in which case `withTransaction` will abort
+  the transaction and return to its caller.
+
+### Applications are responsible for passing ClientSession for operations within a transaction
+
+It remains the responsibility of the application to pass a ClientSession to all operations that should be included in a
+transaction. With regard to `withTransaction`, applications are free to execute any operations within the callback,
+irrespective of whether those operations are associated with the transaction.
+
+### It is assumed that the callback will not start a new transaction on the ClientSession
+
+Under normal circumstances, the callback should not commit the transaction nor should it start a new transaction. The
+`withTransaction` method will inspect the ClientSession's transaction state after the callback returns and take the most
+sensible course of action; however, it will not detect whether the callback has started a new transaction.
+
+### The callback function may be executed multiple times
+
+The implementation of withTransaction is based on the original examples for
+[Retry Transactions and Commit Operation](https://www.mongodb.com/docs/manual/core/transactions/#retry-transaction-and-commit-operation)
```

Comment:
This is a dead anchor. You could conceivably link to https://www.mongodb.com/docs/manual/core/transactions-in-applications/#core-api instead, but at this point I think this entire section could just be reduced to:

> The callback may be executed any number of times. Drivers are free to encourage their
users to design idempotent callbacks.

Ignore this suggestion if you'd rather not make the change in this PR.

Reply:
Done